### PR TITLE
fix type mismatch of max_retry_delay

### LINKED ISSUE #13086: max_retry_delay should be a timedelta for type hinting
**Apache Airflow version**:

master

**What happened**:

https://github.com/apache/airflow/blob/master/airflow/models/baseoperator.py#L356 --> should be timedelta not datetime

### PR DESCRIPTION
**Pull request by iameugenejo**
closes: #13086

Sorry about no test. I don't have in-depth knowledge of this project to come up one.

I'm just trying to fix what's not tested in the first place and broken.

### CODE REVIEW

**Comment by iameugenejo** [comment]
I honestly do not know where this `max_retry_delay` in `timedelta` type gets reassigned as a `float`. so please close this PR if someone finds the code that does that and fix it there as a permanent solution

**Comment by ashb** [comment]
> I honestly do not know where this `max_retry_delay` in `timedelta` type gets reassigned as a `float`. so please close this PR if someone finds the code that does that and fix it there as a permanent solution

Max retry delay is taken from what ever the DAG author puts in their ctor args

File: airflow/models/taskinstance.py [hunk 1]
```diff
@@ -927,7 +927,7 @@ def next_retry_datetime(self):
             delay_backoff_in_seconds = min(modded_hash, timedelta.max.total_seconds() - 1)
             delay = timedelta(seconds=delay_backoff_in_seconds)
             if self.task.max_retry_delay:
-                delay = min(self.task.max_retry_delay, delay)
+                delay = timedelta(seconds=min(self.task.max_retry_delay, delay.total_seconds()))
```
Comment (ashb) [inline airflow/models/taskinstance.py:930]: This change isn't needed -- task.max_retry_delay should be a timedelta)

We should check this in the ctor of BaseOperator, and if what is passed is not a timedleta then do:

```python
    if not isinstance(max_retry_delay):
        max_retry_delay = timedelta(seconds=max_retry_delay)
```

etc.

Could you add a test to tests/models/test_baseoperator.py that checks this upgrade behaviour?

**Review by ashb — CHANGES_REQUESTED** [review]

**Comment by iameugenejo** [comment]
> > I honestly do not know where this `max_retry_delay` in `timedelta` type gets reassigned as a `float`. so please close this PR if someone finds the code that does that and fix it there as a permanent solution
> 
> Max retry delay is taken from what ever the DAG author puts in their ctor args

That's what I thought, and I have every dag with the max_retry_delay set with timedelta value, and the scheduler still fails to start because this value gets converted to float at some point sometimes (not everytime).

**Comment by ashb** [comment]
Closed in favour of https://github.com/apache/airflow/pull/14436

**Pull request closed without merging**